kyo-combinators/shared/src/main/scala/kyo/Combinators.scala

```diff
@@ -77,10 +77,12 @@ extension [A, S](effect: A < S)
       */
     def repeat(schedule: Schedule)(using Flat[A], Frame): A < (S & Async) =
         Loop(schedule) { schedule =>
-            schedule.next.map { (delay, nextSchedule) =>
-                effect.delayed(delay).andThen(Loop.continue(nextSchedule))
-            }.getOrElse {
-                effect.map(Loop.done)
+            Clock.now.map { now =>
```

Comment:
Is it useful to fetch the clock once outside the loop to avoid pulling from the local multiple times?